bot/exts/utilities/color.py

```diff
@@ -0,0 +1,229 @@
+import colorsys
+import json
+import pathlib
+import random
+from io import BytesIO
+
+from PIL import Image, ImageColor
+from discord import Embed, File
+from discord.ext import commands
+from discord.ext.commands.errors import BadArgument
+from rapidfuzz import process
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+THUMBNAIL_SIZE = (80, 80)
+
+
+class Colour(commands.Cog):
+    """Cog for the Colour command."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        with open(pathlib.Path("bot/resources/utilities/ryanzec_colours.json")) as f:
+            self.colour_mapping = json.load(f)
+
+    async def send_colour_response(self, ctx: commands.Context, rgb: tuple[int, int, int]) -> None:
+        """Create and send embed from user given colour information."""
+        r, g, b = rgb
+        name = self._rgb_to_name(rgb)
+        if name == "No match found":
+            name = None
+
+        try:
+            colour_or_color = ctx.invoked_parents[0]
+        except IndexError:
+            colour_or_color = "colour"
+
+        colour_mode = ctx.invoked_with
+        if colour_mode == "random":
+            input_colour = "random"
+        elif colour_mode in ("colour", "color"):
+            input_colour = rgb
+        else:
+            input_colour = ctx.args[2:][0]
+
+        if colour_mode not in ("name", "hex", "random", "color", "colour"):
+            colour_mode = colour_mode.upper()
+        else:
+            colour_mode = colour_mode.title()
+
+        desc = f"{colour_mode} information for `{name or input_colour}`."
+        colour_embed = Embed(
+            title=colour_or_color.title(),
+            description=desc,
+            colour=int(f"{r:02x}{g:02x}{b:02x}", 16)
+        )
+        colour_conversions = self.get_colour_conversions(rgb)
+        for colour_space, value in colour_conversions.items():
+            colour_embed.add_field(
+                name=colour_space,
+                value=f"`{value}`",
+                inline=True
+            )
+
+        thumbnail = Image.new("RGB", THUMBNAIL_SIZE, color=rgb)
+        buffer = BytesIO()
+        thumbnail.save(buffer, "PNG")
+        buffer.seek(0)
+        thumbnail_file = File(buffer, filename="colour.png")
+
+        colour_embed.set_thumbnail(url="attachment://colour.png")
+
+        await ctx.send(file=thumbnail_file, embed=colour_embed)
+
+    @commands.group(aliases=("color",), invoke_without_command=True)
+    async def colour(self, ctx: commands.Context, *, extra: str) -> None:
+        """User initiated command to create an embed that displays colour information."""
+        if ctx.invoked_subcommand is None:
+            try:
+                extra_colour = ImageColor.getrgb(extra)
+                await self.send_colour_response(ctx, extra_colour)
+            except ValueError:
+                await invoke_help_command(ctx)
```

Comment:
Nitpick: You can just return early to ignore the indentation on `ctx.invoked_subcommand is None`.

Reply:
I'm not sure I understand what you mean by that, where would the `return` statement go and what indentation would be changed?

Reply:
```py
if  not ctx.invoked_subcommand:
    return

try:
    extra_colour = ImageColor.getrgb(extra)
    await self.send_colour_response(ctx, extra_colour)
except ValueError:
    await invoke_help_command(ctx)
```

Reply:
Tested it out as written above, it did not work. Keeping the code as below:
```py
    @commands.group(aliases=("color",), invoke_without_command=True)
    async def colour(self, ctx: commands.Context, *, extra: str) -> None:
        """User initiated command to create an embed that displays colour information."""
        if ctx.invoked_subcommand is None:
            try:
                extra_colour = ImageColor.getrgb(extra)
                await self.send_colour_response(ctx, extra_colour)
            except ValueError:
                await invoke_help_command(ctx)
```

Reply:
Tested it out as written above, it did not work. Keeping the code as below:
```py
    @commands.group(aliases=("color",), invoke_without_command=True)
    async def colour(self, ctx: commands.Context, *, extra: str) -> None:
        """User initiated command to create an embed that displays colour information."""
        if ctx.invoked_subcommand is None:
            try:
                extra_colour = ImageColor.getrgb(extra)
                await self.send_colour_response(ctx, extra_colour)
            except ValueError:
                await invoke_help_command(ctx)
```